server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -586,34 +587,55 @@ private ExtensionsSettings readFromExtensionsYml(Path filePath) throws IOExcepti
-                        (Collection<HashMap<String, ?>>) extensionMap.get("dependencies")
-                    );
-                    for (HashMap<String, ?> dependency : extensionDependencies) {
-                        extensionDependencyList.add(
-                            new ExtensionDependency(
-                                dependency.get("uniqueId").toString(),
-                                Version.fromString(dependency.get("version").toString())
-                            )
+                try {
+                    // checking to see whether any required fields are missing from extension.yml file or not
+                    String[] requiredFields = {
+                        "name",
+                        "uniqueId",
+                        "hostAddress",
+                        "port",
+                        "version",
+                        "opensearchVersion",
+                        "minimumCompatibleVersion" };
+                    List<String> missingFields = Arrays.stream(requiredFields)
+                        .filter(field -> !extensionMap.containsKey(field))
+                        .collect(Collectors.toList());
+                    if (!missingFields.isEmpty()) {
+                        throw new IOException("Extension is missing these required fields : " + missingFields);
+                    }
+
+                    // Parse extension dependencies
+                    List<ExtensionDependency> extensionDependencyList = new ArrayList<ExtensionDependency>();
+                    if (extensionMap.get("dependencies") != null) {
+                        List<HashMap<String, ?>> extensionDependencies = new ArrayList<>(
+                            (Collection<HashMap<String, ?>>) extensionMap.get("dependencies")
                         );
+                        for (HashMap<String, ?> dependency : extensionDependencies) {
+                            extensionDependencyList.add(
+                                new ExtensionDependency(
+                                    dependency.get("uniqueId").toString(),
+                                    Version.fromString(dependency.get("version").toString())
+                                )
+                            );
+                        }
                     }
+
+                    // Create extension read from yml config
+                    readExtensions.add(
+                        new Extension(
+                            extensionMap.get("name").toString(),
+                            extensionMap.get("uniqueId").toString(),
+                            extensionMap.get("hostAddress").toString(),
+                            extensionMap.get("port").toString(),
+                            extensionMap.get("version").toString(),
+                            extensionMap.get("opensearchVersion").toString(),
+                            extensionMap.get("minimumCompatibleVersion").toString(),
```

Comment:
Is it time to introduce an extension configuration class?

Reply:
Hi @dblock,

Sorry for the delayed reply, could you please explain a bit more what can be done make extension configuration class ?

Thanks and regards
Varun

Reply:
> Is it time to introduce an extension configuration class?

We do, and the lines quoted above are part of its constructor.

This awkward middle parsing is the result of switching from Jackson (which had concepts of classes/objects) to SnakeYAML, which by default produces a map which we're extracting information from to produce a class.  That PR (#5768) resulted in the bug we're currently fixing.

There are probably ways to have SnakeYAML read directly into a class, but I'm not sure how robust that is to missing data or if it's as CVE-plagued as Jackson was or if it really is exactly what we're doing, hidden inside a dependency.  @ryanbogan did you look into that?

Reply:
It is possible to read directly into a class, but it requires that same `ReflectPermission` to do so that we don't want in core.  That's why I chose to go with this approach.